COVID19App/app/Program.cs

```diff
@@ -21,8 +23,17 @@ static void Main()
 
             var netProvider = new CovidDataProvider();
             var provider = new SQLiteDataProvider();
+            var cacheSystem = new DatabaseCache();
+            cacheSystem.Attach(provider);
 
-            provider.InsertCountryData(netProvider.GetCountryData());
+            try
+            {
+                cacheSystem.CountryInfoList = netProvider.GetCountryData().ToList();
+            }
+            catch (ArgumentException)
+            {
+                //Console.WriteLine("Internet Connection Problem");
```

Comment:
Ar trebui un messagebox or something, nu stiu sigur cum, ca alea parca poti sa le pui doar in interiorul unui form